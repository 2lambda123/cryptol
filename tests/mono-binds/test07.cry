module test07 where

test : {a} a
test = bar
  where
  foo : {b} b -> b
  foo x = x

  // bar depends only on closed things (foo has a complete type signature)
  bar = foo zero
