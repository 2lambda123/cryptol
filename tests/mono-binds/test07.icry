:set debug=on
:set mono-binds=off
:l test07.cry
:set mono-binds=on
:l test07.cry
