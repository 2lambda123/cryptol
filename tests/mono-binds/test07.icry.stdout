Loading module Cryptol
Loading module Cryptol
Loading module test07
module test07
import Cryptol
/* Not recursive */
test07::test : {a} a
test07::test = \{a} ->
               bar a
               where
                 /* Not recursive */
                 foo : {b} b -> b
                 foo = \{b} (x : b) -> x
                 
                 /* Not recursive */
                 bar : {b} b
                 bar = \{b} -> foo b (zero b)
                 
               

Loading module Cryptol
Loading module test07
module test07
import Cryptol
/* Not recursive */
test07::test : {a} a
test07::test = \{a} ->
               bar a
               where
                 /* Not recursive */
                 foo : {b} b -> b
                 foo = \{b} (x : b) -> x
                 
                 /* Not recursive */
                 bar : {b} b
                 bar = \{b} -> foo b (zero b)
                 
               

